src/script/components/LexicalInput/hooks/useIsFocused.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {useLayoutEffect, useState} from 'react';
+
+import {useLexicalComposerContext} from '@lexical/react/LexicalComposerContext';
+import {mergeRegister} from '@lexical/utils';
+import {BLUR_COMMAND, COMMAND_PRIORITY_NORMAL, FOCUS_COMMAND} from 'lexical';
+
+export const useIsFocused = () => {
+  const [editor] = useLexicalComposerContext();
+  const [hasFocus, setHasFocus] = useState(() => editor.getRootElement() === document.activeElement);
+
+  useLayoutEffect(() => {
+    return mergeRegister(
+      editor.registerCommand(
+        FOCUS_COMMAND,
+        () => {
+          setHasFocus(true);
+          return false;
+        },
+        COMMAND_PRIORITY_NORMAL,
+      ),
+      editor.registerCommand(
+        BLUR_COMMAND,
+        () => {
+          setHasFocus(false);
+          return false;
+        },
+        COMMAND_PRIORITY_NORMAL,
+      ),
+    );
```

Comment:
to avoid any potential memory leak we should unregister commands when the component unmounts.

```
const handleFocus = () => {
      setHasFocus(true);
    };
```

    const handleBlur = () => {
      setHasFocus(false);
    };
```
 return () => {
      editor.unregisterCommand(FOCUS_COMMAND, handleFocus);
      editor.unregisterCommand(BLUR_COMMAND, handleBlur);
    };
```